auth/auth.c

```diff
@@ -449,13 +449,17 @@ static void gip_auth_exchange_rsa(struct work_struct *work)
 					     work_exchange_rsa);
 	struct gip_auth_pkt_host_secret pkt = {};
 	u8 random[GIP_AUTH_RANDOM_LEN * 2];
-	u8 pms[GIP_AUTH_SECRET_LEN];
+	u8 *pms;
 	int err;
 
 	memcpy(random, auth->random_host, sizeof(auth->random_host));
 	memcpy(random + sizeof(auth->random_host), auth->random_client,
 	       sizeof(auth->random_client));
 
+	pms = kmalloc(GIP_AUTH_SECRET_LEN, GFP_KERNEL);
+	if (!pms)
+		return -ENOMEM;
+
 	/* get random premaster secret */
 	get_random_bytes(pms, sizeof(pms));
```

Comment:
All `sizeof(pms)` are now incorrect. Before it was the size of the array (in bytes), now it's the size of a pointer. Easiest is to replace all occurances with GIP_AUTH_SECRET_LEN.

Reply:
Of course, you're right. Interesting that it worked even with the incorrect size passed to the auth functions...